core/src/Kokkos_LogicalSpaces.hpp

```diff
@@ -0,0 +1,433 @@
+
+  void* impl_allocate(const char* arg_label, const size_t arg_alloc_size,
+                      const size_t arg_logical_size = 0,
+                      Kokkos::Tools::SpaceHandle arg_handle =
+                          Kokkos::Tools::make_space_handle(name())) const {
+    return underlying_allocator.impl_allocate(arg_label, arg_alloc_size,
+                                              arg_logical_size, arg_handle);
+  }
+  void impl_deallocate(const char* arg_label, void* const arg_alloc_ptr,
+                       const size_t arg_alloc_size,
+                       const size_t arg_logical_size = 0,
+                       const Kokkos::Tools::SpaceHandle arg_handle =
+                           Kokkos::Tools::make_space_handle(name())) const {
+    underlying_allocator.impl_deallocate(
+        arg_label, arg_alloc_ptr, arg_alloc_size, arg_logical_size, arg_handle);
+  }
+};
+}  // namespace Experimental
+}  // namespace Kokkos
+
+//----------------------------------------------------------------------------
+
+namespace Kokkos {
+
+namespace Impl {
+
+template <typename BaseSpace, typename DefaultExecutionSpace, class Namer,
+          typename OtherSpace>
+struct MemorySpaceAccess<Kokkos::Experimental::LogicalMemorySpace<
+                             BaseSpace, DefaultExecutionSpace, Namer, true>,
+                         OtherSpace> {
+  enum { assignable = MemorySpaceAccess<BaseSpace, OtherSpace>::assignable };
+  enum { accessible = MemorySpaceAccess<BaseSpace, OtherSpace>::accessible };
+  enum { deepcopy = MemorySpaceAccess<BaseSpace, OtherSpace>::deepcopy };
+};
+
+template <typename BaseSpace, typename DefaultExecutionSpace, class Namer,
+          typename OtherSpace>
+struct MemorySpaceAccess<OtherSpace,
+                         Kokkos::Experimental::LogicalMemorySpace<
+                             BaseSpace, DefaultExecutionSpace, Namer, true>> {
+  enum { assignable = MemorySpaceAccess<OtherSpace, BaseSpace>::assignable };
+  enum { accessible = MemorySpaceAccess<OtherSpace, BaseSpace>::accessible };
+  enum { deepcopy = MemorySpaceAccess<OtherSpace, BaseSpace>::deepcopy };
+};
+
+template <typename BaseSpace, typename DefaultExecutionSpace, class Namer>
+struct MemorySpaceAccess<Kokkos::Experimental::LogicalMemorySpace<
+                             BaseSpace, DefaultExecutionSpace, Namer, true>,
+                         Kokkos::Experimental::LogicalMemorySpace<
+                             BaseSpace, DefaultExecutionSpace, Namer, true>> {
+  enum { assignable = true };
+  enum { accessible = true };
+  enum { deepcopy = true };
+};
+
+template <typename BaseSpace, typename DefaultExecutionSpace, class Namer>
+struct MemorySpaceAccess<Kokkos::AnonymousSpace,
+                         Kokkos::Experimental::LogicalMemorySpace<
+                             BaseSpace, DefaultExecutionSpace, Namer, true>> {
+  enum { assignable = true };
+  enum { accessible = true };
+  enum { deepcopy = true };
+};
```

Comment:
Covered by https://github.com/kokkos/kokkos/blob/87fd4d123946cfe5ffa7357818f6343c4861658f/core/src/Kokkos_AnonymousSpace.hpp#L86-L91

Reply:
That is a fantastic catch